circularBuffer.hpp

```diff
@@ -0,0 +1,116 @@
+#ifndef circularBuffer_
+#define circsularBuffer_
+
+#define BufferSize 5             //circular buffer size 
+
+//**********************************************************************************
+// Author: Sascha Heinisch
+// Date:   28.11.2019
+// 
+// Class: Declarartion of circular buffer class. 
+//
+//**********************************************************************************
+template <class T> class circularBuffer
+{
+  public:
+
+    static T buffer[BufferSize];
+    static size_t head;              //head of circular buffer
+    static size_t tail;              //tail of circular buffer
+    static bool   full;              //buffer full if true
```

Comment:
Do you think `full` needs to be a member variable?

Reply:
I don't get your point. Is this the wrong place?

Reply:
The question is why do you need a `full` member variable. Can it be implemented without it?

Reply:
Yes it can be implemented without, but on the cost of one entry.

Reply:
Should I do so?

Reply:
No it's not necessary to change, thank you